src/ComponentEvent.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2015 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+import { ComponentEventConstructor } from "./types";
+
+export class _ComponentEvent<PROPS extends Record<string, any>, TYPE extends string = string, THIS = any> {
```

Comment:
It is strange to use underscore in the class name.

It is even strange to expose underscore as the type of the trigger method.

Reply:
You're right. It is mean to be named like it's an actual ComponentEvent but it can be named elseway.